PR 3/3 - fix(bookmark): remove bookmark count (#431)

martaag18 opened the pull request:
**This PR 431 is part of the User Story 390 → Como alumn@, puedo dar/quitar bookmark a un reto**

**-> This branch was created from feature/432-show-user-bookmarks.**

**What this PR does:**
- Removes the display of the bookmark counter (bookmarks_count) in the UI.
- Updates the components (ChallengeCardComponent, ChallengeHeaderComponent) and their related tests to eliminate all usage of the timesBookmarked value.
- Focuses the logic solely on whether the challenge is bookmarked or not — not how many times.

![image](https://github.com/user-attachments/assets/888579c6-9c22-4064-a4fd-bd3a0590f2cc)

**How to test it:**

1. Log in as any user.
2. Go to any challenge list or challenge detail page.
3. Confirm that the bookmark icon is shown, but there is no numerical counter next to it.
4. The icon still correctly shows bookmarked/not bookmarked status and toggles as expected.

**To consider:**
- This change has no impact on backend logic — only the front-end display has been modified.
- Since there’s no backend endpoint for timesBookmarked, this data was unnecessary and is now removed.

martaag18 committed:
Add 'bookmarked' property to Challenge model and bind to challenge card

martaag18 committed:
feat: remove bookmark count display from ChallengeCard component

martaag18 committed:
feat: remove bookmark cound display from ChallengeHeader component

martaag18 committed:
test: remove bookmarks_count in tests

martaag18 committed:
docs: add changelog entry for features 429, 432

AlexMafaluno commented on src/app/modules/challenge/components/challenge-header/challenge-header.component.ts:
```diff
@@ -183,4 +184,30 @@ export class ChallengeHeaderComponent implements OnInit {
+          this.isBookmarked = response.bookmarked
+        },
+        error: error => {
+          console.error('Error removing bookmark:', error)
```
is console.error necessary?

martaag18 replied:
I kept the console.error for now because it’s the only way we’ll notice if the bookmark add/remove operation fails, especially while we don’t have user-facing error handling in place.

AlexMafaluno commented on src/app/modules/challenge/components/challenge-header/challenge-header.component.ts:
```diff
@@ -183,4 +184,30 @@ export class ChallengeHeaderComponent implements OnInit {
+          this.isBookmarked = response.bookmarked
+        },
+        error: error => {
+          console.error('Error adding bookmark:', error)
```
here too

martaag18 replied:
I kept the console.error for now because it’s the only way we’ll notice if the bookmark add/remove operation fails, especially while we don’t have user-facing error handling in place.

AlexMafaluno commented:
Hi Marta, taking a quick look at the functionality I noticed an error in the tooltip:
![image](https://github.com/user-attachments/assets/af15085b-69f4-46ce-9f5f-5e148cc497b0)

martaag18 commented:
> Hi Marta, taking a quick look at the functionality I noticed an error in the tooltip:

@AlexMafaluno it is all good now, I've fixed the problem in my first PR (429) as it was requested by Ivan. Since this branch was created from 429, I’ve updated it accordingly and included all the latest changes here as well.

AlexMafaluno commented:
Alright @martaag18 , if this is treated as technical debt and you've addressed it in PR429, I'm fine with approving this.
